File: src/main/java/com/aladdinworks8/service/ServiceContractService.java
package com.aladdinworks8.service;

import java.util.List;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.jpa.domain.Specification;
import org.springframework.http.ResponseEntity;

import com.aladdinworks8.domain.ServiceContract;
import com.aladdinworks8.dto.ServiceContractDTO;
import com.aladdinworks8.dto.ServiceContractSearchDTO;
import com.aladdinworks8.dto.ServiceContractPageDTO;
import com.aladdinworks8.dto.ServiceContractConvertCriteriaDTO;
import com.aladdinworks8.service.GenericService;
import com.aladdinworks8.dto.common.RequestDTO;
import com.aladdinworks8.dto.common.ResultDTO;
import java.util.List;
import java.util.Optional;





public interface ServiceContractService extends GenericService<ServiceContract, Integer> {

	List<ServiceContract> findAll();

	ResultDTO addServiceContract(ServiceContractDTO serviceContractDTO, RequestDTO requestDTO);

	ResultDTO updateServiceContract(ServiceContractDTO serviceContractDTO, RequestDTO requestDTO);

    Page<ServiceContract> getAllServiceContracts(Pageable pageable);

    Page<ServiceContract> getAllServiceContracts(Specification<ServiceContract> spec, Pageable pageable);

	ResponseEntity<ServiceContractPageDTO> getServiceContracts(ServiceContractSearchDTO serviceContractSearchDTO);
	
	List<ServiceContractDTO> convertServiceContractsToServiceContractDTOs(List<ServiceContract> serviceContracts, ServiceContractConvertCriteriaDTO convertCriteria);

	ServiceContractDTO getServiceContractDTOById(Integer serviceContractId);







}

File: aladdinworks8_ui/src/services/ServiceContractService.js
import http from "../http-common"; 

class ServiceContractService {
  getAllServiceContracts(searchDTO) {
    console.log(searchDTO)
    return this.getRequest(`/serviceContract/serviceContracts`, searchDTO);
  }

  get(serviceContractId) {
    return this.getRequest(`/serviceContract/${serviceContractId}`, null);
  }

  findByField(matchData) {
    return this.getRequest(`/serviceContract?field=${matchData}`, null);
  }

  addServiceContract(data) {
    return http.post("/serviceContract/addServiceContract", data);
  }

  update(data) {
  	return http.post("/serviceContract/updateServiceContract", data);
  }
  
  uploadImage(data,serviceContractId) {
  	return http.postForm("/serviceContract/uploadImage/"+serviceContractId, data);
  }




	postRequest(url, data) {
		return http.post(url, data);
      };

	getRequest(url, params) {
        return http.get(url, {
        	params: params
        });
    };

}

export default new ServiceContractService();
